benches/merkle_tree.rs

```diff
@@ -0,0 +1,21 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use miden_crypto::{merkle::MerkleTree, Word};
+use rand_utils::rand_array;
+pub use winter_math::fields::f64::BaseElement as Felt;
+
+static BATCH_SIZES: [usize; 3] = [65536, 131072, 262144];
+
+pub fn merkle_tree_construction(c: &mut Criterion) {
+    let mut merkle_group = c.benchmark_group("MerkleTree");
+
+    for size in &BATCH_SIZES {
+        let data: Vec<Word> = (0..*size).map(|_| rand_array::<Felt, 4>()).collect();
+
+        merkle_group.bench_with_input(BenchmarkId::new("construction", size), &data, |b, i| {
+            b.iter(|| MerkleTree::new(i))
+        });
```

Comment:
The current form is ok, but did you also consider using batched inputs? Its a good strategy for cases where we want to initialize distinct sets and measure the performance, such as in this one.

[Bencher::iter_batched](https://docs.rs/criterion/0.4.0/criterion/struct.Bencher.html#method.iter_batched)

Also, possibly unrelated to this PR, but we could benefit from measurements of the other routines too (insert, update, etc)

Reply:
Hey, I did not! I based this on the code from winterfell [ref](https://github.com/0xPolygonMiden/crypto/pull/58#issuecomment-1431967114). Thanks for the link I will have a look at it :)

Reply:
hey, I went over the docs and `iter_batched` doesn't seem to add any interesting feature, just extra runtime to generate fresh data for each run. Am I missing something?